Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -335,9 +335,9 @@ Uri GetAbsoluteUri(Uri relativeUri)
 			var query = parseUri["q"].Value;
 			var fragment = parseUri["f"].Value;
 
-			Element item = CurrentItem;
+			Element item = ignoreCurrentItem ? CurrentItem.Parent : CurrentItem;
```

Comment:
I think it is not quite right. If the idea is to ignore the current item, then it parent will not always be its root.
The best solution is to remove all items from the list between the first and the last. 
`if (list.Count > 2) list.RemoveRange(1, list.Count - 2);`